.github/workflows/test.yml

```diff
@@ -0,0 +1,355 @@
+              repo: context.repo.repo,
+              head: `${context.repo.owner}:${{ github.ref }}`
+            });
+
+            return !!pullR.data.length;
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # https://github.com/actions/checkout/releases/tag/v2.3.4
+        if: ${{ github.event_name == 'push' }}
+      - uses: dorny/paths-filter@b2feaf19c27470162a626bd6fa8438ae5b263721 # https://github.com/dorny/paths-filter/releases/tag/v2.10.2
+        if: ${{ !(steps.has-pr.outputs.result == 'true' && github.event_name == 'push') }}
+        id: path-filter
+        with:
+          filters: |
+            build-handsontable-umd:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/e2e/**'
+              - 'test/scripts/**'
+            build-handsontable-es-cjs:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'wrappers/**'
+            test-handsontable-unit:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/unit/**'
+            test-handsontable-types:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/types/**'
+            test-handsontable-walkontable:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/scripts/**'
+            test-handsontable-e2e:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/e2e/**'
+              - 'test/scripts/**'
+            test-handsontable-e2e-min:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'test/e2e/**'
+              - 'test/scripts/**'
+            test-angular:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'wrappers/angular/**'
+            test-react:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'wrappers/react/**'
+            test-vue:
+              - '.config/**'
+              - 'src/**'
+              - 'scripts/**'
+              - 'wrappers/vue/**'
+
+  build-handsontable-umd:
+    name: "[BUILD] Handsontable: UMD"
+    runs-on: ubuntu-latest
+    needs: [ check-scope ]
+    if: |
+      needs.check-scope.outputs.run-all == 'true' ||
+      (needs.check-scope.outputs.skip-all == 'false' &&
+      needs.check-scope.outputs.build-handsontable-umd == 'true')
+    steps:
+      - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # https://github.com/actions/checkout/releases/tag/v2.3.4
+      - name: Use Node.js ${{ matrix.node-version }}
+        uses: actions/setup-node@25316bbc1f10ac9d8798711f44914b1cf3c4e954 # https://github.com/actions/setup-node/releases/tag/v2.4.0
+        with:
+          node-version: 15.9.0
```

Comment:
WDYT about extracting the node version as env ([docs](https://docs.github.com/en/actions/reference/context-and-expression-syntax-for-github-actions#example-setting-an-environment-variable))? The node version is repeated 10 times right now.